### ISSUE TITLE
Emit more documentation regarding validation

### ISSUE THREAD

**Issue by dominikh** [interagent/prmd#218]
prmd currently only supports a small subset of possible validation in its generation of documentation. Some requirements, such as a string's length are embedded in the description, but a lot aren't. For example, none of the validations for arrays are documented: minItems, maxItems, uniqueItems, or those of the individual items.

I don't think that it's feasible to cram all of those into the description, so I would propose an extra section a la "Validation", that lists each of them in a list or table. Special consideration will be required for arrays  because they can have arbitrary levels of depth.

**Comment by geemus** [comment]
Yeah, good point. Simply adding it to the description was "good enough" initially, but I agree that as you get into more complex/thorough validations it starts to feel inadequate quickly. A separate validation sections sounds good for this. Definitely will take a bit of thought and perhaps prototyping to sort out how best to include this in things.
